crates/engine/tree/src/tree/mod.rs

```diff
@@ -673,6 +690,34 @@ where
         Ok(())
     }
 
+    fn on_downloaded_block(&mut self, block: SealedBlockWithSenders) -> Option<TreeEvent> {
+        let block_hash = block.hash();
+        let lowest_buffered_ancestor = self.lowest_buffered_ancestor_or(block_hash);
+        if let Some(status) =
```

Comment:
Is `status` used later?
```suggestion
        if let Some(_) =
```